File: bindings/java/src/test/java/org/wickra/compile/GoldenTest.java
package org.wickra.compile;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

// The cross-language golden invariant seen from Java: the same spec yields a
// byte-identical manifest across calls, and those bytes are what every other
// binding produces too, because the whole compiler lives once in the Rust core
// and this binding forwards its JSON verbatim.
class GoldenTest {
    @Test
    void compileIsByteIdenticalAcrossCalls() {
        try (Compiler a = new Compiler(); Compiler b = new Compiler()) {
            assertEquals(a.command(CompilerTest.COMPILE_CMD), b.command(CompilerTest.COMPILE_CMD));
        }
    }
}

File: bindings/go/golden_test.go
package wickra

// The cross-language golden invariant seen from Go: the same spec yields a
// byte-identical manifest across calls, and those bytes are what every other
// binding produces too, because the whole compiler lives once in the Rust core
// and this binding forwards its JSON verbatim.

import "testing"

func TestCompileByteIdenticalAcrossCalls(t *testing.T) {
	a := New()
	defer a.Close()
	b := New()
	defer b.Close()

	ra, err := a.Command(compileCmd)
	if err != nil {
		t.Fatal(err)
	}
	rb, err := b.Command(compileCmd)
	if err != nil {
		t.Fatal(err)
	}
	if ra != rb {
		t.Fatalf("expected byte-identical output, got:\n a: %s\n b: %s", ra, rb)
	}
}
